app/main.py

```diff
@@ -1,2 +1,67 @@
+from typing import Any, Hashable, List, Optional
+from fractions import Fraction
+
+
+class Node:
+    def __init__(self, key: Hashable, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key)
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.load_factor = Fraction(2, 3)
+        self.table: List[Optional[List[Node]]] = [None] * self.capacity
+
+    def _resize(self) -> None:
+        old_table = self.table
+        self.capacity *= 2
+        self.table = [None] * self.capacity
+        self.size = 0
+        for bucket in old_table:
+            if bucket:
+                for node in bucket:
+                    self.__setitem__(node.key, node.value)
+
+    def _hash_index(self, key: Hashable) -> int:
+        return hash(key) % self.capacity
+
+    def _find_node(
+        self,
+        bucket: Optional[List[Node]],
+        key: Hashable
+    ) -> Optional[Node]:
+        if bucket:
+            for node in bucket:
+                if node.key == key:
+                    return node
+        return None
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size / self.capacity > self.load_factor:
+            self._resize()
+
+        index = self._hash_index(key)
+        if self.table[index] is None:
+            self.table[index] = []
+
+        node = self._find_node(self.table[index], key)
+        if node:
+            node.value = value
+        else:
+            self.table[index].append(Node(key, value))
```

Comment:
you are doing a list that has a list of Nodes - this solution is less effective in terms of complexity. Looking for element will take O(n) and we are looking for the best performance.

Reply:
so you need to have only one list of Nodes. each node must have unique index, collisions must be taken care of and table needs to be resized with recalculation of indexes - review learning materials about how it needs to be done